File: app/csharp/src/ProductionManagement.Infrastructure/Rest/Controllers/InventoryController.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using ProductionManagement.Application.Port.In;
using ProductionManagement.Domain.Models.Inventory;
using ProductionManagement.Infrastructure.Rest.Dto;

namespace ProductionManagement.Infrastructure.Rest.Controllers;

/// <summary>
/// 在庫照会 Controller（Input Adapter）
/// </summary>
[ApiController]
[Route("api/inventory")]
[Tags("Inventory")]
public class InventoryController : ControllerBase
{
    private readonly IInventoryUseCase _inventoryUseCase;

    public InventoryController(IInventoryUseCase inventoryUseCase)
    {
        _inventoryUseCase = inventoryUseCase;
    }

    /// <summary>
    /// 在庫一覧の取得
    /// </summary>
    /// <param name="itemCode">品目コード（フィルター）</param>
    /// <param name="locationCode">場所コード（フィルター）</param>
    /// <param name="status">在庫ステータス（フィルター）</param>
    [HttpGet]
    [ProducesResponseType(typeof(IReadOnlyList<StockResponse>), StatusCodes.Status200OK)]
    public async Task<IActionResult> GetInventory(
        [FromQuery] string? itemCode,
        [FromQuery] string? locationCode,
        [FromQuery] StockStatus? status)
    {
        var query = new InventoryQuery(itemCode, locationCode, status);
        var stocks = await _inventoryUseCase.GetInventoryAsync(query);
        return Ok(stocks.Select(StockResponse.From).ToList());
    }

    /// <summary>
    /// 在庫サマリーの取得
    /// </summary>
    [HttpGet("summary")]
    [ProducesResponseType(typeof(IReadOnlyList<InventorySummaryResponse>), StatusCodes.Status200OK)]
    public async Task<IActionResult> GetInventorySummary()
    {
        var summaries = await _inventoryUseCase.GetInventorySummaryAsync();
        return Ok(summaries.Select(InventorySummaryResponse.From).ToList());
    }

    /// <summary>
    /// 在庫不足品目の取得
    /// </summary>
    [HttpGet("shortage")]
    [ProducesResponseType(typeof(IReadOnlyList<InventorySummaryResponse>), StatusCodes.Status200OK)]
    public async Task<IActionResult> GetShortageItems()
    {
        var items = await _inventoryUseCase.GetShortageItemsAsync();
        return Ok(items.Select(InventorySummaryResponse.From).ToList());
    }
}

File: app/java/src/main/java/com/example/production/infrastructure/in/rest/controller/InventoryController.java
package com.example.production.infrastructure.in.rest.controller;

import com.example.production.application.port.in.InventoryUseCase;
import com.example.production.domain.model.inventory.Stock;
import com.example.production.infrastructure.in.rest.dto.StockResponse;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.Parameter;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * 在庫 Controller（Input Adapter）
 */
@RestController
@RequestMapping("/api/inventory")
@Tag(name = "inventory", description = "在庫 API")
@RequiredArgsConstructor
public class InventoryController {

    private final InventoryUseCase useCase;

    @GetMapping
    @Operation(summary = "在庫一覧の取得")
    public ResponseEntity<List<StockResponse>> getInventory(
            @Parameter(description = "品目コード")
            @RequestParam(required = false) String itemCode,
            @Parameter(description = "場所コード")
            @RequestParam(required = false) String locationCode,
            @Parameter(description = "在庫状態")
            @RequestParam(required = false) String status) {

        InventoryUseCase.InventoryQuery query = new InventoryUseCase.InventoryQuery(
                itemCode, locationCode, status);

        List<Stock> stocks = useCase.getInventory(query);
        return ResponseEntity.ok(stocks.stream()
                .map(StockResponse::from)
                .toList());
    }

    @GetMapping("/location/{locationCode}")
    @Operation(summary = "場所別在庫一覧の取得")
    public ResponseEntity<List<StockResponse>> getStocksByLocation(
            @Parameter(description = "場所コード", required = true)
            @PathVariable String locationCode) {

        List<Stock> stocks = useCase.getStocksByLocation(locationCode);
        return ResponseEntity.ok(stocks.stream()
                .map(StockResponse::from)
                .toList());
    }

    @GetMapping("/item/{itemCode}")
    @Operation(summary = "品目別在庫一覧の取得")
    public ResponseEntity<List<StockResponse>> getStocksByItem(
            @Parameter(description = "品目コード", required = true)
            @PathVariable String itemCode) {

        List<Stock> stocks = useCase.getStocksByItem(itemCode);
        return ResponseEntity.ok(stocks.stream()
                .map(StockResponse::from)
                .toList());
    }

    @GetMapping("/{locationCode}/{itemCode}")
    @Operation(summary = "在庫の取得")
    public ResponseEntity<StockResponse> getStock(
            @Parameter(description = "場所コード", required = true)
            @PathVariable String locationCode,
            @Parameter(description = "品目コード", required = true)
            @PathVariable String itemCode) {

        Stock stock = useCase.getStock(locationCode, itemCode);
        return ResponseEntity.ok(StockResponse.from(stock));
    }
}
